src/config.rs

```diff
@@ -0,0 +1,35 @@
+use std::collections::HashMap;
+use std::path::Path;
+
+/// Configuration of `bors` loaded from a config file.
+#[derive(serde::Deserialize, Debug, Clone)]
+pub struct Config {
+    pub github: GithubConfig,
+    pub repo: HashMap<String, GithubRepositoryConfig>,
+}
+
+impl Config {
+    /// Loads the config from a TOML file.
+    pub fn from_toml_file(path: &Path) -> std::io::Result<Self> {
+        let data = std::fs::read(path)?;
+        Ok(toml::from_slice(&data)?)
+    }
+}
+
+#[derive(serde::Deserialize, Debug, Clone)]
+pub struct GithubConfig {
+    /// GitHub personal access token.
+    pub access_token: String,
+    /// GitHub OAuth application client ID
+    pub app_client_id: String,
+    /// GitHub OAuth application client secret
+    pub app_client_secret: String,
```

Comment:
We could use the https://docs.rs/secrecy/latest/secrecy/ crate to avoid accidentally leaking secrets.

Reply:
Good idea, I added it.